internal/saucecloud/cloud.go

```diff
@@ -486,21 +486,40 @@ func (r *CloudRunner) remoteArchiveFiles(project interface{}, files []string, sa
 }
 
 // FetchJUnitReports retrieves junit reports for the given result and all of its
-// attempts.
-func (r *CloudRunner) FetchJUnitReports(res *result) {
+// attempts. Can use the given artifacts to avoid unnecessary API calls.
+func (r *CloudRunner) FetchJUnitReports(res *result, artifacts []report.Artifact) {
 	if !report.IsArtifactRequired(r.Reporters, report.JUnitArtifact) {
 		return
 	}
 
+	var junitArtifact *report.Artifact
+	for _, artifact := range artifacts {
+		if strings.HasSuffix(artifact.FilePath, junit.FileName) {
+			junitArtifact = &artifact
+			break
+		}
+	}
+
 	for i := range res.attempts {
 		attempt := &res.attempts[i]
 
-		content, err := r.JobService.GetJobAssetFileContent(
-			context.Background(),
-			attempt.ID,
-			junit.FileName,
-			res.job.IsRDC,
-		)
+		var content []byte
+		var err error
+
+		// If this is the last attempt, we can use the given junit artifact to
+		// avoid unnecessary API calls.
+		if i == len(res.attempts)-1 && junitArtifact != nil {
+			content, err = os.ReadFile(junitArtifact.FilePath)
+			log.Info().Msg("Using local JUnit report") // TODO remove me
```

Comment:
I actually decided to keep that around, as it can be useful for troubleshooting. Just converted that to a debug statement, since it's generally not something the user needs to see all the time 👍